### PR TITLE
🐛 Stop audio on tab change

### LINKED ISSUE #3: [Sounds] Audio does not stop when changing tabs
Hi,

If I play a sound, then click on the "Quotes" tab, the sound does not stop, even though I am not in the Sounds tab anymore.

Then if I go back to the "Sounds" tab and play any sound, the previous sound is not muted, and the second sound plays over the first one.

I think this is related to the fact that a new `Audio` instance is created each time a `SoundItem` component mounts (via `useEffect`), but the `Audio` instance from the previous incarnation of that component is never paused or cleared. We should do something about it when the component unmounts.

### PR DESCRIPTION
**Pull request by thaisguigon**
Fixes #3

### CODE REVIEW

File: src/components/SoundItem.tsx [hunk 1]
```diff
@@ -28,30 +28,61 @@ const useSoundPlayer = (props: SoundItemProps) => {
...
+      },
+    ];
+
+    eventListeners.forEach(({ type, listener }) =>
+      audio.addEventListener(type, listener),
+    );
+
+    return () => {
+      audio.pause();
+      eventListeners.forEach(({ type, listener }) =>
```
Comment (cbellino) [inline src/components/SoundItem.tsx]: Not gonna hide it, i'm really not a fan how this way of handling the listeners but i agree we have to do it. This is part of why i didn't implement this until now :'(

Comment (cbellino) [reply src/components/SoundItem.tsx]:
```javascript
If we play around with tuples we can have something like this, what do you think?

const eventListeners: [string, () => void][] = [
      [
        "canplaythrough",
        () => {
          setLoaded(true);
          setAudio(audio);
          setDuration(audio.duration);
        },
      ],
      [
        "error",
        () => {
          console.error(`Error loading: "${sound.audioUrl}".`);
          setLoaded(false);
        },
      ],
      [
        "play",
        () => {
          setPaused(false);
          onPlay(sound.id);
        },
      ],
      [
        "ended",
        () => {
          setPaused(true);
          onPause(sound.id);
        },
      ],
      [
        "pause",
        () => {
          setPaused(true);
          onPause(sound.id);
        },
      ],
      [
        "timeupdate",
        () => {
          setProgress(audio.currentTime / audio.duration);
        },
      ],
    ];

    eventListeners.forEach(args => audio.addEventListener(...args));

    return () => {
      audio.pause();
      eventListeners.forEach(args => audio.removeEventListener(...args));
    };
```

**Comment by cbellino** [comment]
Woops, did a commit without noticing with WIP stuff, reverted 😱

File: src/components/SoundItem.tsx (hunk 1 above)
Comment (thaisguigon) [reply src/components/SoundItem.tsx]: Would you rather use an object?

```javascript
const eventListeners = {
  canplaythrough: () => {
    setLoaded(true);
    setAudio(audio);
    setDuration(audio.duration);
  },
  error: () => {
    console.error(`Error loading: "${sound.audioUrl}".`);
    setLoaded(false);
  },
  play: () => {
    setPaused(false);
    onPlay(sound.id);
  },
  ended: () => {
    setPaused(true);
    onPause(sound.id);
  },
  pause: () => {
    setPaused(true);
    onPause(sound.id);
  },
  timeupdate: () => {
    setProgress(audio.currentTime / audio.duration);
  },
};
```

Comment (thaisguigon) [reply src/components/SoundItem.tsx]: (The type would be something like: `{ [type: string]: () => void }` I think?)

**Comment by thaisguigon** [comment]
Closing this as the code changed so much, I opened another PR to fix this: https://github.com/cbellino/hubside-quotes/pull/9

**Pull request closed without merging**